src/domain/accounts/index.ts

```diff
@@ -22,11 +23,34 @@ export * from "./primitives"
 const UserIdRegex =
   /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
 
+// device id format from AppCheck: 1:72279297366:android:35666807ae916c5aa75af7
+const DeviceIdRegex = /^[0-9]+:[0-9]+:[a-z]+:[0-9a-z]+$/i
+
+export const checkedToDeviceAccountUserId = (userId: string) => {
+  if (userId.match(/^[0-9]+:[0-9]+:[a-z]+:[0-9a-z]+$/i)) {
+    return userId as UserId
+  }
+
+  return new InvalidUserId(userId)
+}
+
 export const checkedToUserId = (userId: string): UserId | ValidationError => {
-  if (!userId.match(UserIdRegex)) {
-    return new InvalidUserId(userId)
+  // kratosUserId is a uuid v4
+  if (userId.match(UserIdRegex)) {
```

Comment:
even the `UserIdRegex` regex is giving hindsight that there is some inconsistency here with https://github.com/GaloyMoney/galoy/pull/2071/files#diff-f365a6e339ebf8a923ca76401012e4718a4ee3e4b8a2bd460ededbdf41467dc4R31